db/write_batch.cc

```diff
@@ -1659,29 +1684,34 @@ class MemTableInserter : public WriteBatch::Handler {
       auto* cfd =
           static_cast_with_check<ColumnFamilyHandleImpl>(cf_handle)->cfd();
       if (!cfd->is_delete_range_supported()) {
-        return Status::NotSupported(
+        ret_status = Status::NotSupported(
```

Comment:
Can we restructure the code to not have ret_status = Status::Oops(); return ret_status?

I assume that is necessary for the status check code, but it looks really screwy and folks are not going to understand the reasons or nuance.

Reply:
OK, for now I put `ret_status.PermitUncheckedError()` here. The root cause is `SeekToColumnFamily()` uses both a `bool` and a `Status` to indicate its outcome, but for the return statements here, checking the `bool` is sufficient (`true` implies `Status::OK()`). We should refactor `SeekToColumnFamily()` to return only a `Status`, though I didn't want to expand this PR's scope.